--- a/net/minecraft/world/level/storage/TagValueOutput.java
+++ b/net/minecraft/world/level/storage/TagValueOutput.java
@@ -17,15 +_,21 @@
     private final ProblemReporter problemReporter;
     private final DynamicOps<Tag> ops;
     private final CompoundTag output;
+    public final HolderLookup.@org.jspecify.annotations.Nullable Provider lookup;
 
     TagValueOutput(ProblemReporter p_407091_, DynamicOps<Tag> p_407083_, CompoundTag p_409593_) {
+        this(p_407091_, p_407083_, p_409593_, null);
+    }
+
+    TagValueOutput(ProblemReporter p_407091_, DynamicOps<Tag> p_407083_, CompoundTag p_409593_, HolderLookup.@org.jspecify.annotations.Nullable Provider lookup) {
         this.problemReporter = p_407091_;
         this.ops = p_407083_;
         this.output = p_409593_;
+        this.lookup = lookup;
     }
 
     public static TagValueOutput createWithContext(ProblemReporter p_409887_, HolderLookup.Provider p_409738_) {
-        return new TagValueOutput(p_409887_, p_409738_.createSerializationContext(NbtOps.INSTANCE), new CompoundTag());
+        return new TagValueOutput(p_409887_, p_409738_.createSerializationContext(NbtOps.INSTANCE), new CompoundTag(), p_409738_);
     }
 
     public static TagValueOutput createWithoutContext(ProblemReporter p_405875_) {
